block.go

```diff
@@ -350,6 +353,89 @@ type Divider struct {
 	// empty
 }
 
+type EquationBlock struct {
+	BasicBlock
+	Equation Equation `json:"equation"`
+}
+
+type Equation struct {
+	Expression string `json:"expression"`
+}
+
+type BreadcrumbBlock struct {
+	BasicBlock
+	Breadcrumb Breadcrumb `json:"breadcrumb"`
+}
+
+type Breadcrumb struct {
+	// empty
+}
+
+type ColumnBlock struct {
+	BasicBlock
+	Column Column `json:"column"`
+}
+
+type Column struct {
+	// Children should at least have 1 block when appending.
+	Children []Block `json:"children"`
+}
+
+type ColumnListBlock struct {
+	BasicBlock
+	ColumnList ColumnList `json:"column_list"`
+}
+
+type ColumnList struct {
+	// Children can only contain column blocks
+	// Children should have at least 2 blocks when appending.
+	Children []Block `json:"children"`
+}
+
+// NOTE: will only be returned by the API. Cannot be created by the API.
+// https://developers.notion.com/reference/block#link-preview-blocks
+type LinkPreviewBlock struct {
+	BasicBlock
+	LinkPreview LinkPreview `json:"link_preview"`
+}
+
+type LinkPreview struct {
+	URL string `json:"url"`
+}
+
+type LinkToPageBlock struct {
+	BasicBlock
+	LinkToPage LinkToPage `json:"link_to_page"`
+}
+
+type LinkToPage struct {
+	Type   string `json:"type"`
+	PageID string `json:"page_id"`
+}
+
+type TemplateBlock struct {
+	BasicBlock
+	Template Template `json:"template"`
+}
+
+type Template struct {
+	Text     []RichText `json:"text"`
+	Children []Block    `json:"children,omitempty"`
+}
+
+type SyncedBlock struct {
+	BasicBlock
+	Synced Synced `json:"synced_block"`
+}
+
+type Synced struct {
+	// SyncedFrom is nil for the original block.
+	SyncedFrom *struct {
```

Comment:
I'm not sure that using the unnamed struct would be convenient for library users.
Let's move `SyncedForm` to a separated struct. What do you think?

Reply:
Sure this sounds good.

A possible idea to make the library more convenient for users could be to have constructor functions. My assumption is that users would only really need to create their own blocks when updating/creating blocks in Notion. In that case I believe all the blocks need is the type and the "properties" like "paragraph" or "heading_1". In that case it could be useful to give a function like:
```go
func NewParagraphBlock(text []RichText, children []Block) *ParagraphBlock {
  return &ParagraphBlock{
    BasicBlock {
      Type : "paragraph"
    },
    Text : text,
    Children: children,
  }
}
```
Curious what you think.
In this case the user wouldn't need to worry about first, setting the correct type, and second, the layout of the block like with the anonymous struct.

This could allow for something like:
```go
blk := NewColumnListBlock([]Block{
  NewColumnBlock([]Block{
    NewHeadingBlock(...SomeRichText...),
  }),
  NewColumnBlock([]Block{
    NewParagraphBlock(...SomeText..., nil)
  }),
})
```
(Not sure if nil will work or not in terms of marshalling).